evento/src/views/Explore/index.js

```diff
@@ -4,21 +4,37 @@ import EventCard from '../../components/EventCard';
 import './Explore.css';
 
 class Explore extends Component {
-	componentWillMount() {
-		this.setState({ events: [] });
+	constructor(props) {
+		super(props);
+		this.state = { events: [] };
+	}
 
+	componentDidMount() {
 		fetch('/events')
 		.then(response => response.json())
 		.then(events => {
 			this.setState({ events: events });
 		});
 	}
+	
+	isVisible(event, filterQuery) {
```

Comment:
If the filtering function is implemented in MainPage or SearchBar, those components would have to have knowledge of events and events are only fetched in Explore / My Events. So I'd say it would be better if Explore / My Events could decide which fields the MainPage filter would use. That way MainPage doesn't rely on API and changes in API could be implemented in Explore without changing MainPage. On the other hand there would still be a bit duplicate code in Explore and My Events.

Implementing category filtering makes this a bit harder.